Not really sure what's happened, all the Travis builds on Python 3.x failed with:

``` pytb
Traceback (most recent call last):
  File "setup.py", line 81, in <module>
    cmdclass={'test': PyTest},
  File "/opt/python/3.3.5/lib/python3.3/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/opt/python/3.3.5/lib/python3.3/distutils/dist.py", line 930, in run_commands
    self.run_command(cmd)
  File "/opt/python/3.3.5/lib/python3.3/distutils/dist.py", line 948, in run_command
    cmd_obj.ensure_finalized()
  File "/opt/python/3.3.5/lib/python3.3/distutils/cmd.py", line 107, in ensure_finalized
    self.finalize_options()
  File "setup.py", line 15, in finalize_options
    self.test_args = ['--verbose', 'tests/']
AttributeError: can't set attribute
```

but `tox -e py34` works fine for me. :confused:

--- a/marshmallow/marshalling.py
+++ b/marshmallow/marshalling.py
@@ -21,6 +21,8 @@ __all__ = [
     'Unmarshaller',
 ]
 
+# Key used for field-level validation errors on nested fields
+FIELD = '_field'
 
 class ErrorStore(object):
 
@@ -68,6 +70,8 @@ class ErrorStore(object):
             # Warning: Mutation!
             if isinstance(err.messages, dict):
                 errors[field_name] = err.messages
+            elif isinstance(errors.get(field_name), dict):
+                errors[field_name].setdefault(FIELD, []).extend(err.messages)
             else:
                 errors.setdefault(field_name, []).extend(err.messages)
             # When a Nested field fails validation, the marshalled data is stored

--- a/tests/test_schema.py
+++ b/tests/test_schema.py
@@ -1222,6 +1222,23 @@ class TestNestedSchema:
         _, errors = schema.load({'inner': 1})
         assert errors['inner']['_schema'] == ['Invalid input type.']
 
+    # regression test for https://github.com/marshmallow-code/marshmallow/issues/298
+    def test_all_errors_on_many_nested_field_with_validates_decorator(self):
+        class Inner(Schema):
+            req = fields.Field(required=True)
+
+        class Outer(Schema):
+            inner = fields.Nested(Inner, many=True)
+
+            @validates('inner')
+            def validates_inner(self, data):
+                raise ValidationError('not a chance')
+
+        outer = Outer()
+        _, errors = outer.load({'inner': [{}]})
+        assert 'inner' in errors
+        assert '_field' in errors['inner']
+
     def test_missing_required_nested_field(self):
         class Inner(Schema):
             inner_req = fields.Field(required=True, error_messages={'required': 'Oops'})

